### DISCUSSION TITLE
REST API 6.2

### DISCUSSION THREAD

**Discussion by drakub** [YetiForceCompany/YetiForceCRM#15422] [General]
Hello, I have been facing challenges in configuring REST API in 6.2. This was an easy task with 6.1 but 6.2 has a few changes.
I have realized, after creating a web service (REST API), when creating a web service username, I don't have options to provide a username and password as was the case with 6.1. The system generates login (which I assume is the username) as **name_cloud@yetiforce.com**. I get 401 (Unauthorized) error when using this username. In a second guess, I used the username of the user profile (in this case, Administrator) and the corresponding password to login but still to no avail. Can someone help me here? What am I doing wrong with webservice configuration in yetiforce 6.2? attached are screen shots of my API client.

I have also enabled webservice in config/Api.php

**AUTHORIZATION**
![image](https://user-images.githubusercontent.com/19745222/140873924-e8d85a5f-4973-482f-8474-182bfa518677.png)

**HEADERS**
![image](https://user-images.githubusercontent.com/19745222/140874086-9331fe55-44f2-4796-a0d1-1cd65e020e72.png)

**BODY**
![image](https://user-images.githubusercontent.com/19745222/140874172-f16a14cd-23f9-4d83-ae4a-4e036216e898.png)

**Comment by rskrzypczak** [comment]
https://doc.yetiforce.com/api/?urls.primaryName=Webservice%20Standard
Api authorization is the same in v6.2 and v6.1.
The only difference is basically the change of address

```
rewrite
CRM_URL/webservice/RestApi/Users/Login
CRM_URL/webservice/RestApi/Accounts/RecordRelatedList/117/Contacts
without rewrite
CRM_URL/webservice.php?_container=RestApi&module=Users&action=Login
CRM_URL/webservice.php?_container=RestApi&module=Accounts&action=RecordRelatedList&record=117&param=Contacts
```
Check if you have correctly set rewrite on your server

**Comment by drakub** [comment]
This is the same URL I am calling but the error still persists. My rewrite
rules are correct

**Reply by mariuszkrzaczkowski** [reply]
Show raw form of requests and responses

**Reply by drakub** [reply]
**Here is my PH CURL REQUEST**
```

<?php

$curl = curl_init();

curl_setopt_array($curl, array(
  CURLOPT_URL => 'http://IP_ADDRESS/webservice/RestApi/Users/Login',
  CURLOPT_RETURNTRANSFER => true,
  CURLOPT_ENCODING => '',
  CURLOPT_MAXREDIRS => 10,
  CURLOPT_TIMEOUT => 0,
  CURLOPT_FOLLOWLOCATION => true,
  CURLOPT_HTTP_VERSION => CURL_HTTP_VERSION_1_1,
  CURLOPT_CUSTOMREQUEST => 'POST',
  CURLOPT_POSTFIELDS => array('userName' => 'ura_cloud@yetiforce.com','password' => 'rest_test'),
  CURLOPT_HTTPHEADER => array(
    'Content-Type: application/json',
    'x-api-key: 7GyfX7x7JWhAbQh7gjBHQ1wZtH78n2VY',
    'Authorization: Basic UG9ydGFsUmVzdDpQb3J0YWxSZXN0'
  ),
));

$response = curl_exec($curl);

curl_close($curl);
echo $response;
```

**Comment by drakub** [comment]
All those I shared on the screenshot
